src/cli/run.rs

```diff
@@ -195,6 +205,9 @@ pub async fn get_task_env<'p>(
     lock_file_derived_data: &mut LockFileDerivedData<'p>,
     environment: &Environment<'p>,
 ) -> miette::Result<HashMap<String, String>> {
+    // Make sure the system requirements are met
+    verify_current_platform_has_required_virtual_packages(environment)?;
```

Comment:
This will only check the requirements after possibly having run other tasks already. I think we should check the system requirements when creating the task graph.

Reply:
Ah yes good point.